pkg/meta/redis.go

```diff
@@ -1224,6 +1228,9 @@ func (m *redisMeta) doUnlink(ctx Context, parent Ino, name string) syscall.Errno
 		if pattr.Typ != TypeDirectory {
 			return syscall.ENOTDIR
 		}
+		if (pattr.Flags&FlagAppend) != 0 || (pattr.Flags&FlagImmutable) != 0 {
+			return syscall.EPERM
+		}
```

Comment:
should check itself flags in [redis.go#L1249](https://github.com/juicedata/juicefs/blob/b3c8473d9f7dd308798f86d39d211b13d8b5d6c6/pkg/meta/redis.go#L1249)
 
```go
if (attr.Flags&FlagAppend) != 0 || (attr.Flags&FlagImmutable) != 0 {
        return syscall.EPERM
}
```